Sources/SwiftProtobuf/TextFormatEncoder.swift

```diff
@@ -14,165 +14,261 @@
 
 import Foundation
 
+private let asciiSpace = UInt8(ascii: " ")
+private let asciiColon = UInt8(ascii: ":")
+private let asciiComma = UInt8(ascii: ",")
+private let asciiMinus = UInt8(ascii: "-")
+private let asciiBackslash = UInt8(ascii: "\\")
+private let asciiDoubleQuote = UInt8(ascii: "\"")
+private let asciiZero = UInt8(ascii: "0")
+private let asciiOpenCurlyBracket = UInt8(ascii: "{")
+private let asciiCloseCurlyBracket = UInt8(ascii: "}")
+private let asciiOpenSquareBracket = UInt8(ascii: "[")
+private let asciiCloseSquareBracket = UInt8(ascii: "]")
+private let asciiNewline = UInt8(ascii: "\n")
+private let asciiLowerB = UInt8(ascii: "b")
+private let asciiLowerF = UInt8(ascii: "f")
+private let asciiLowerN = UInt8(ascii: "n")
+private let asciiLowerR = UInt8(ascii: "r")
+private let asciiLowerT = UInt8(ascii: "t")
+private let asciiLowerV = UInt8(ascii: "v")
+
 /// TextFormatEncoder has no public members.
-public final class TextFormatEncoder {
-    var text: String = ""
+internal struct TextFormatEncoder {
+    private var data = [UInt8]()
```

Comment:
`Data` has some interesting locking behavior over the `NSData` that it wraps internally, I found, by digging into the implementation. Doing byte-by-byte operations on it is slower than it needs to be :(